lib/domain/usecase/create_room_interactor.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/room/create_room_failed.dart';
+import 'package:fluffychat/domain/app_state/room/create_room_success.dart';
+import 'package:fluffychat/domain/app_state/room/set_room_avatar_failed.dart';
+import 'package:fluffychat/domain/app_state/room/set_room_avatar_success.dart';
+import 'package:fluffychat/domain/model/contact/new_room_informations.dart';
+import 'package:fluffychat/domain/usecase/set_room_avatar_interactor.dart';
+import 'package:matrix/matrix.dart';
+
+class CreateRoomInteractor {
+  final setRoomAvatarInteractor = getIt.get<SetRoomAvatarInteractor>();
+
+  Stream<Either<CreateRoomFailed, CreateRoomSuccess>> execute(
+      Client matrixClient, NewRoomInformations newRoomInformations) async* {
+    try {
+      final roomId = await matrixClient.createGroupChat(
+        groupName: newRoomInformations.groupName,
+        invite: newRoomInformations.invite,
+        enableEncryption: newRoomInformations.enableEncryption,
+        preset: newRoomInformations.createRoomPreset,
+      );
+
+      if (roomId.isNotEmpty) {
+        yield Right(
+            CreateRoomSuccess(event: CreateRoomSuccessEventType.createRoomSuccess, roomId: roomId));
+
+        if (newRoomInformations.avatar != null) {
+          await for (final Either<SetRoomAvatarFailed, SetRoomAvatarSuccess> result
+              in setRoomAvatarInteractor.execute(
+                  matrixClient, roomId, newRoomInformations.avatar)) {
+            if (result.isRight()) {
+              yield Right(CreateRoomSuccess(
+                  event: CreateRoomSuccessEventType.setAvatarSuccess, roomId: roomId));
+            } else if (result.isLeft()) {
+              throw result.fold((l) => l.exception, (r) => r);
+            }

```

Comment:
you can see, to combine some actors, it is not an easy work. Still good, but I want to make it clearer for the reader